lib/stat_tracker.rb

```diff
@@ -17,9 +17,33 @@ def self.from_csv(locations)
   end
 
   def self.read_from_file(file)
-    file_open = CSV.open(file)
+    CSV.parse(File.read(file), headers: true)
+  end
+
+# ************* LeagueStatistics *************
+
+  def group_by(data, key, value)
+    hash = {}
+    data = self.instance_variable_get(data)
+    data.each do |row|
+      if hash[row[key]]
+        hash[row[key]] << row[value]
+      else
+        hash[row[key]] = [row[value]]
+      end
+    end
+    hash
+  end
```

Comment:
Not sure if this PR has a test for this in the test file. 

Looks good though and I appreciate how useful I think this will be for other methods.

Am I correct then by assuming if I call group_by in my methods, I'll have to use `:@teams` instead of just `@teams`? I think I'll have to do more research on what the `:` does for my own understanding.